src/main/java/com/exercise/masters/controller/BookController.java

```diff
@@ -0,0 +1,71 @@
+/**
+ * 
+ */
+package com.exercise.masters.controller;
+
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.exercise.masters.exceptions.RecordNotFoundException;
+import com.exercise.masters.model.Book;
+import com.exercise.masters.services.BookService;
+
+/**
+ * @author michaeldelacruz
+ *
+ */
+
+@RestController
+public class BookController {
+	
+	@Autowired
+	private BookService bookService;
+	
+	@GetMapping("/author/{authorId}/books")
+	public ResponseEntity<Page<Book>> findAllBooks(
```

Comment:
Can use the exact same name in pageable (sort=id,asc&page=0&size=2), then you can use it like

@GetMapping("/authors/{authorId}/books")
public Page getAllBooksByAuthorId(@PathVariable (value = "authorId") Long authorId,
Pageable pageable) {

}